src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -91,25 +91,38 @@ parseTypeSynonymDeclaration =
                          <*> (indented *> equals *> noWildcards parsePolyType)
 
 parseValueDeclaration :: TokenParser Declaration
-parseValueDeclaration = do
+parseValueDeclaration = P.try parseValueDeclarationWithIdentFirst <|> parseValueDeclarationWithInfixOperator
```

Comment:
We need to be careful that this doesn't affect the quality of the error messages too much...